### PR TITLE
Add support of TEST_TAGS in python tests

### PR DESCRIPTION
**Pull request by jsoriano**
## What does this PR do?

Add a `tag(tag)` decorator that skips a test if the tag is not included
in the comma-separated list of `TEST_TAGS` environment variable.
This offers an initial support for the similar implementation added for
mage in #16937.

## Why is it important?

#16937 was intended for cloud tests, where we only have go-based
integration tests. But we have found similar needings in some services
like oracle, where we need an image that we cannot distribute. In that
case we need a similar approach to run the test only in our Jenkins
instances, where the image is available.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] ~~I have made corresponding changes to the documentation~~
- [x] ~~I have made corresponding change to the default configuration files~~
- [x] ~~I have added tests that prove my fix is effective or that my feature works~~

## How to test this PR locally

* Tag a python test with `@metricbeat.tag('sometag')`.
* Check that the test is skipped if `TEST_TAGS` is not used.
* Check that the test is executed if `TEST_TAGS` environment variable includes `sometag` (e.g: `TEST_TAGS=sometag` or `TEST_TAGS=sometag,foo`.

## Related issues

- Required by #16833 to enable tests only in Jenkins.
- Extends #16937.
- Could be useful in #15891 if we add system tests for cloud features.

## Use cases

- Run certain tests that require some application or credentials to be available only where they are.

### CODE REVIEW

File: metricbeat/tests/system/metricbeat.py [hunk 1]
```diff
@@ -139,3 +140,26 @@ def parameterized_with_supported_versions(base_class):
     variants = supported_versions(versions_path)
     decorator = parameterized_class(['COMPOSE_ENV'], variants)
     decorator(base_class)
+
+
+def tag(tag):
```
Comment (sayden) [inline metricbeat/tests/system/metricbeat.py]: This is then used with `@metricbeat.tag`?

Comment (jsoriano) [reply metricbeat/tests/system/metricbeat.py]: Yes, you need to import `metricbeat` for that, but I think we do it in all metricbeat tests.

Comment (jsoriano) [reply metricbeat/tests/system/metricbeat.py]: Well, you need to pass a tag as parameter, e.g:
```
@metricbeat.tag('oracle')
def test_something(self):
    ...
```

**Comment by jsoriano** [comment]
@sayden could you take another look after moving the code to libbeat? Thanks!

**Pull request merged**